docs/hooks.md

```diff
@@ -0,0 +1,108 @@
+# Hooks
+
+Cucumber provides a number of hooks which allow us to run blocks at various points in the Cucumber test cycle. You can define them anywhere in your support or step definition layers, using the methods `Before` and `After`. Hooks are global by default, meaning they run for every scenario in your
+features. If you want them to run for just certain scenarios, you need to tag
+those scenarios and then use a [tagged hook](hooks.md#tagged-hooks).
+
+## Scenario hooks
+
+`Before` hooks will be run before the first step of each scenario. They will run in the same order of which they are registered.
+
+{% codetabs name="Ruby", type="rb" -%}
+Before do
+  # Do something before each scenario.
+end
+{%- language name="Java", type="java" -%}
+  {% raw %}
+  @Before
+  public void beforeCallingScenario() {
+    System.out.println("*********** About to start the scenario.");
+  }
+  {% endraw %}
+{%- endcodetabs %}
+
+or like this:
+
+{% codetabs name="Ruby", type="rb" -%}
+Before do |scenario|
```

Comment:
We discourage this kind of logging - Cucumber's own formatters take care of that. Instead, show something more useful you can do with a Scenario object, like embedding something into the reports, like an image (typically from a screenshot) or text.

A better example is to use the scenario